File: Final Project/ClientApp/src/app/Park.ts
// To parse this data:
//
//   import { Convert, Post } from "./file";
//
//   const post = Convert.toPost(json);

export interface Park {
    total?: string;
    limit?: string;
    start?: string;
    data?: Datum[];
}

export interface Datum {
    id?: string;
    url?: string;
    fullName?: string;
    parkCode?: string;
    description?: string;
    latitude?: string;
    longitude?: string;
    latLong?: string;
    activities?: Activity[];
    topics?: Activity[];
    states?: string;
    contacts?: Contacts;
    entranceFees?: Entrance[];
    entrancePasses?: Entrance[];
    fees?: any[];
    directionsInfo?: string;
    directionsUrl?: string;
    operatingHours?: OperatingHour[];
    addresses?: Address[];
    images?: Image[];
    weatherInfo?: string;
    name?: string;
    designation?: string;
}

export interface Activity {
    id?: string;
    name?: string;
}

export interface Address {
    postalCode?: string;
    city?: string;
    stateCode?: string;
    line1?: string;
    type?: string;
    line3?: string;
    line2?: string;
}

export interface Contacts {
    phoneNumbers?: PhoneNumber[];
    emailAddresses?: EmailAddress[];
}

export interface EmailAddress {
    description?: string;
    emailAddress?: string;
}

export interface PhoneNumber {
    phoneNumber?: string;
    description?: string;
    extension?: string;
    type?: string;
}

export interface Entrance {
    cost?: string;
    description?: string;
    title?: string;
}

export interface Image {
    credit?: string;
    title?: string;
    altText?: string;
    caption?: string;
    url?: string;
}

export interface OperatingHour {
    exceptions?: Exception[];
    description?: string;
    standardHours?: Hours;
    name?: string;
}

export interface Exception {
    exceptionHours?: Hours;
    startDate?: Date;
    name?: string;
    endDate?: Date;
}

export interface Hours {
    wednesday?: Day;
    monday?: Day;
    thursday?: Day;
    sunday?: Day;
    tuesday?: Day;
    friday?: Day;
    saturday?: Day;
}

export enum Day {
    AllDay = "All Day",
    Closed = "Closed",
}

// Converts JSON strings to/from your types
export class Convert {
    public static toPost(json: string): Park {
        return JSON.parse(json);
    }

    public static postToJson(value: Park): string {
        return JSON.stringify(value);
    }
}

File: Final Project/Models/Park.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Final_Project.Models
{
    public class Park
    {
        public string total { get; set; }
        public string limit { get; set; }
        public string start { get; set; }
        public Datum[] data { get; set; }
    }

    public class Datum
    {
        public string id { get; set; }
        public string url { get; set; }
        public string fullName { get; set; }
        public string parkCode { get; set; }
        public string description { get; set; }
        public string latitude { get; set; }
        public string longitude { get; set; }
        public string latLong { get; set; }
        public Activity[] activities { get; set; }
        public Topic[] topics { get; set; }
        public string states { get; set; }
        public Contacts contacts { get; set; }
        public Entrancefee[] entranceFees { get; set; }
        public object[] entrancePasses { get; set; }
        public object[] fees { get; set; }
        public string directionsInfo { get; set; }
        public string directionsUrl { get; set; }
        public Operatinghour[] operatingHours { get; set; }
        public Address[] addresses { get; set; }
        public Image[] images { get; set; }
        public string weatherInfo { get; set; }
        public string name { get; set; }
        public string designation { get; set; }
    }

    public class Contacts
    {
        public Phonenumber[] phoneNumbers { get; set; }
        public Emailaddress[] emailAddresses { get; set; }
    }

    public class Phonenumber
    {
        public string phoneNumber { get; set; }
        public string description { get; set; }
        public string extension { get; set; }
        public string type { get; set; }
    }

    public class Emailaddress
    {
        public string description { get; set; }
        public string emailAddress { get; set; }
    }

    public class Activity
    {
        public string id { get; set; }
        public string name { get; set; }
    }

    public class Topic
    {
        public string id { get; set; }
        public string name { get; set; }
    }

    public class Entrancefee
    {
        public string cost { get; set; }
        public string description { get; set; }
        public string title { get; set; }
    }

    public class Operatinghour
    {
        public Exception[] exceptions { get; set; }
        public string description { get; set; }
        public Standardhours standardHours { get; set; }
        public string name { get; set; }
    }

    public class Standardhours
    {
        public string wednesday { get; set; }
        public string monday { get; set; }
        public string thursday { get; set; }
        public string sunday { get; set; }
        public string tuesday { get; set; }
        public string friday { get; set; }
        public string saturday { get; set; }
    }

    public class Exception
    {
        public Exceptionhours exceptionHours { get; set; }
        public string startDate { get; set; }
        public string name { get; set; }
        public string endDate { get; set; }
    }

    public class Exceptionhours
    {
    }

    public class Address
    {
        public string postalCode { get; set; }
        public string city { get; set; }
        public string stateCode { get; set; }
        public string line1 { get; set; }
        public string type { get; set; }
        public string line3 { get; set; }
        public string line2 { get; set; }
    }

    public class Image
    {
        public string credit { get; set; }
        public string title { get; set; }
        public string altText { get; set; }
        public string caption { get; set; }
        public string url { get; set; }
    }

}

